tests/unit/test_logging.py

```diff
@@ -0,0 +1,184 @@
+import logging
+
+import _pytest.logging
+import pytest
+
+from tmt.log import (DebugLevelFilter, Logger, QuietnessFilter,
+                     VerbosityLevelFilter)
+
+from . import assert_log
+
+
+def test_sanity(caplog: _pytest.logging.LogCaptureFixture, root_logger: Logger) -> None:
+    root_logger.print('this is printed')
+    root_logger.debug('this is a debug message')
+    root_logger.verbose('this is a verbose message')
+    root_logger.info('this is just an info')
+    root_logger.warn('this is a warning')
+    root_logger.fail('this is a failure')
+
+    assert_log(caplog, details_key='this is printed', levelno=logging.INFO)
+    assert_log(caplog, details_key='this is a debug message', levelno=logging.DEBUG)
+    assert_log(caplog, details_key='this is a verbose message', levelno=logging.INFO)
+    assert_log(caplog, details_key='this is just an info', levelno=logging.INFO)
+    assert_log(caplog, details_key='warn', details_value='this is a warning', levelno=logging.WARN)
+    assert_log(
+        caplog,
+        details_key='fail',
+        details_value='this is a failure',
+        levelno=logging.ERROR)
+
+
+def test_creation(caplog: _pytest.logging.LogCaptureFixture, root_logger: Logger) -> None:
+    logger = Logger.create()
+    assert logger._logger.name == 'tmt'
+
+    actual_logger = logging.Logger('3rd-party-app-logger')
+    logger = Logger.create(actual_logger)
+    assert logger._logger is actual_logger
+
+
+def test_descend(caplog: _pytest.logging.LogCaptureFixture, root_logger: Logger) -> None:
+    deeper_logger = root_logger.descend().descend().descend()
+
+    deeper_logger.print('this is printed')
+    deeper_logger.debug('this is a debug message')
+    deeper_logger.verbose('this is a verbose message')
+    deeper_logger.info('this is just an info')
+    deeper_logger.warn('this is a warning')
+    deeper_logger.fail('this is a failure')
+
+    assert_log(caplog, details_key='this is printed', levelno=logging.INFO)
+    assert_log(caplog, details_key='this is a debug message', levelno=logging.DEBUG)
+    assert_log(caplog, details_key='this is a verbose message', levelno=logging.INFO)
+    assert_log(caplog, details_key='this is just an info', levelno=logging.INFO)
+    assert_log(caplog, details_key='warn', details_value='this is a warning', levelno=logging.WARN)
+    assert_log(
+        caplog,
+        details_key='fail',
+        details_value='this is a failure',
+        levelno=logging.ERROR)
+
+
+@pytest.mark.parametrize(
+    ('logger_verbosity', 'message_verbosity', 'filter_outcome'),
+    [
+        # (
+        #   logger verbosity - corresponds to -v, -vv, -vvv CLI options,
+        #   message verbosity - `level` parameter of `verbosity(...)` call,
+        #   expected outcome of `VerbosityLevelFilter.filter()` - returns integer!
+        # )
+        (0, 0, 1),
+        (1, 0, 1),
+        (2, 0, 1),
+        (3, 0, 1),
+        (4, 0, 1),
```

Comment:
There are no `verbose(level=0)` messages, so perhaps these five can be removed?

Reply:
I included those for the sake of completeness, to venture into the territory of corner cases.

Reply:
Dropped in 6d14224c36569188909070dbdd7f1a5fe7fcbf6f